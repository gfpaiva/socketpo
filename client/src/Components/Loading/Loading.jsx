import React from 'react';

import Loader from 'react-loaders';

import './Loading.scss';

const Loading = ( {...props} ) => (
	<Loader
		{...props}
		type="ball-pulse"
	/>
);

export default Loading;

