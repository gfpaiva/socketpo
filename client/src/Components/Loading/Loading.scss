@import '../../Scss/root';
@import '~loaders.css/src/animations/ball-pulse.scss';

.loader-hidden {
	display: none;
}

.loader-active {
	display: block;
}
