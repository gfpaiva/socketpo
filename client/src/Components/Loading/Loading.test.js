import React from 'react';
import { mount } from 'enzyme';
import Loading from './Loading';

describe('<Loading />', () => {
	it('should mount properly with props', () => {
		const wrapper = mount(
			<Loading test="test"/>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
