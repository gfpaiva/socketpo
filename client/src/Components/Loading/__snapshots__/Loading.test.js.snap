// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should mount properly with props 1`] = `
<Loading
  test="test"
>
  <Loader
    active={true}
    test="test"
    type="ball-pulse"
  >
    <div
      className="loader loader-undefined loader-active"
    >
      <div
        className="loader-inner ball-pulse"
      >
        <div
          key="0"
          style={Object {}}
        />
        <div
          key="1"
          style={Object {}}
        />
        <div
          key="2"
          style={Object {}}
        />
      </div>
    </div>
  </Loader>
</Loading>
`;
